frontend/services/viewer/viewerService/viewerController.tsx

```diff
@@ -0,0 +1,666 @@
+		return this.initialized && !this.pinDropMode && !this.measureMode;
+	}
+
+	public async highlightObjects(params) {
+		if (this.canHighlight) {
+			const ids = params.id ? [params.id] : params.ids;
+			if (ids) {
+				const uniqueIds = Array.from(new Set(ids));
+				if (uniqueIds.length) {
+					await this.highlightAllObjects(
+							params.account,
+							params.model,
+							ids,
+							params.colour,
+							params.multi,
+							params.forceReHighlightt);
+
+					this.emit(VIEWER_EVENTS.HIGHLIGHT_OBJECTS, {account: params.account, model: params.model, uniqueIds });
+					return;
+				}
+			}
+
+			this.clearHighlights();
+		}
+	}
+
+	public unhighlightObjects(params) {
+		this.unhighlightAllObjects(
+				params.account,
+				params.model,
+				params.id ? [params.id] : params.ids
+		);
+	}
+
+	/**
+	 * Settings
+	 */
+
+	public updateViewerSettings(settings) {
+		if (settings) {
+			this.settings = settings;
+			if (this.settings.properties && this.settings.properties.unit) {
+				this.setUnits(this.settings.properties.unit);
+			}
+		}
+	}
+
+	public setUnits(units) {
+		this.units = units;
+
+		if (units === 'mm') {
+			this.convertToM = 0.001;
+		} else if (units === 'ft') {
+			this.convertToM = 0.0032;
+		}
+
+		// Set the units in unity for the measure tool
+		if (this.units) {
+			UnityUtil.setUnits(this.units);
+		}
+	}
+
+	/**
+	 * Nav mode
+	 */
+
+	public setNavigationMode(mode) {
+		this.setNavMode(mode, false);
+	}
+
+	public setNavMode(mode, force) {
+		if (this.currentNavMode !== mode || force) {
+			this.currentNavMode = mode;
+			this.setNavigation(mode);
+		}
+	}
+
+	public navMethodChanged(newNavMode) {
+		this.currentNavMode = newNavMode;
+	}
+
+	public get getNavMode() {
+		return this.currentNavMode;
+	}
+
+	/**
+	 * Fullscreen
+	 */
+
+	public switchFullScreen() {
+		if (!this.fullscreen) {
+			if (this.viewer.hasOwnProperty('mozRequestFullScreen')) {
+			} else if (this.viewer.webkitRequestFullscreen) {
+				this.viewer.webkitRequestFullscreen();
+			}
+
+			this.fullscreen = true;
+		} else {
+			if (document.webkitCancelFullScreen) {
+				document.webkitCancelFullScreen();
+			}
+
+			this.fullscreen = false;
+		}
+	}
+
+	/**
+	 * Pins
+	 */
+
+	public addPin = async ({account, model, id, type, pickedPos: position, norm, colours, viewpoint, isSelected}) => {
+			await this.isViewerReady();
+			this.pins[id] = new Pin({ id, type, position, norm, colors: colours, account, model });
+
+			if (isSelected) {
+				this.selectPin(id);
+			}
+			await this.isModelReady();
+			if (type === 'risk') {
+				await this.dropRiskPin(id, position, norm, colours);
+			} else {
+				await this.dropIssuePin(id, position, norm, colours);
+			}
+	}
+
+	public removePin = ({id}) => {
+		if (this.pins.hasOwnProperty(id)) {
+			this.pins[id].remove();
+			delete this.pins[id];
+		}
+	}
+
+	public removeUnsavedPin() {
+		this.removePin({ id: this.newPinId });
+		this.setPin({ data: null });
+	}
+
+	public highlightPin(id) {
```

Comment:
I don't think we use this? I think the highlight state isn't kept here and we just outright call change color.

I think this can be removed (but check the dependency!)